src/feelpp/benchmarking/reframe/report.py

```diff
@@ -192,68 +256,199 @@ def partialSpeedUp(self):
+
         for t in sorted(self.df_perf['num_tasks'].unique(), reverse=False):
             df_task = self.df_perf[self.df_perf['num_tasks'] == t]
-            fig.add_trace(
-                go.Bar(x=df_task['name'], y=df_task['value'], name=str(t)))
+            df_task.loc[:, 'name'] = df_task['name'].apply(lambda x: x.split('_')[0] if self.partial else x)
+            fig.add_trace(go.Bar(x=df_task['name'], y=df_task['value'], name=str(t)), row=1, col=1)
+
+        # Table (Warning message if only df['name'] without .loc due to views)
+        df_table = self.df_perf
+        if self.partial:
+            df_table.loc[:, 'name'] = df_table['name'].apply(lambda x: x.split('_')[0])
+
+        table = self.plotTable(df_table)
+        fig.add_trace(table, row=2, col=1)
 
-        fig.update_layout(barmode='group', xaxis_tickangle=-45,
-                          title='Performance by step', yaxis_type='log')
+        fig.update_layout(barmode='group', xaxis_tickangle=-45, title='Performance by step', yaxis_type='log', height=800)
         return fig
 
 
     def plotPerformanceByTask(self):
-        fig = px.bar(self.df_perf, x="num_tasks", y="value", title='Performance by task',
-                     color="name", barmode="group", log_y=True, log_x=True)
+        df = self.df_perf.copy()
+        df['name'] = df['name'].apply(lambda x: x.split('_')[0] if self.partial else x)
+
+        fig = px.bar(df, x="num_tasks", y="value", title='Performance by task',
+                    color="name", barmode="group", log_y=True, log_x=True)
+
         return fig
 
 
+    def buildButtons(self, nbCurves):
+        """
+        Builds 'toggle' buttons for displaying regression and optimality curves or not
+        Index = n-th 'fig.add_trace()' call, starting by 0
+        """
+        regIndex = list(range(1, nbCurves-2, 2))
+        optiIndex = list(range(nbCurves-3, nbCurves))
+        menu = [dict(   type="buttons",
+                        buttons=[dict(  label="Display linear regression",
+                                        method="update",
+                                        args=[{"visible":True}, {}, regIndex],
+                                        args2=[{"visible":False}, {}, regIndex]
+                                    ),
+                                dict(   label="Display optimality",
+                                        method="update",
+                                        args=[{"visible":True}, {}, optiIndex],
+                                        args2=[{"visible":False}, {}, optiIndex]
+                                    )],
+                        active=-1,
+                        showactive=True,
+                        x=1,
+                        y=1.06,
+                        xanchor="right",
+                        yanchor="top",
+                        direction="right"
+                    )]
+        return menu
+
+
     def plotSpeedup(self):
```

Comment:
All these plot functions should be refactored with work on #72 on another branch